include/qore/ReferenceHolder.h

```diff
@@ -125,6 +126,13 @@ class SimpleRefHolder {
 public:
    DLLLOCAL SimpleRefHolder() : p(0) {}
    DLLLOCAL SimpleRefHolder(T* p_) : p(p_) {}
+
+   //! move constructor
+   /** @since %Qore 0.9
+   */
+   DLLLOCAL SimpleRefHolder(SimpleRefHolder&& old) : p(std::move(old.p)) {
```

Comment:
this way it's gonna be `deref`ed twice, because `old.p` will be still pointing where it was. You need to assign nullptr to it.

Reply:
ha ha I had that and then removed it, because I assumed that `std::move()` would leave the source empty for simple values, silly me!

Reply:
done